lib/index.js

```diff
@@ -315,6 +316,26 @@ Socket.prototype.maxreconn  = opt('maxreconn');
 Socket.prototype.sndprio    = opt('sndprio');
 Socket.prototype.rcvprio    = opt('rcvprio');
 
+/* chan and rmchan sockopt methods. only relevant for subscription sockets */
+Socket.prototype.chan       = function(){
+  var args = arguments;
+  var channels = args.length;
+  while(--channels > -1) {
```

Comment:
Not if you subtract 1 from your array index, or use post decrement.
```
while (--i > -1) { arr[i]; }
```
compared to:
```
while (i--) { arr[i] }
```